common/src/main/java/com/wynntils/features/user/CombatXPGainMessageFeature.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.notifications.NotificationManager;
+import com.wynntils.mc.event.SetExperienceEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.CharacterManager;
+import com.wynntils.wynn.utils.WynnUtils;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class CombatXPGainMessageFeature extends UserFeature {
+    @Config
+    public float secondDelay = 0.5f;
+
+    private long lastExperienceDisplayTime = 0;
+
+    private float newTickXP = 0;
+    private float lastTickXP = 0;
+    private float trackedPercentage = 0;
+    private int trackedLevel = 0;
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        lastExperienceDisplayTime = 0;
+        newTickXP = 0;
+        lastTickXP = 0;
+        trackedPercentage = 0;
+    }
+
+    @SubscribeEvent
+    public void onExperienceChange(SetExperienceEvent event) {
```

Comment:
```suggestion
    public void onXpChange(SetXpEvent event) {
```

I think we should stick to XP for experience points. We could use "Experience" consistently as well, but that is not as clear and concise, I think. Mixing the two randomly is the least good solution.